rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1299,36 +1166,45 @@ public NodeState getNodeState(String nodeUrl) {
-     */
-    public BooleanWrapper createNodeSource(String nodeSourceName, String infrastructureType,
-            Object[] infrastructureParameters, String policyType, Object[] policyParameters, boolean nodesRecoverable) {
-
+    @Override
+    public BooleanWrapper defineNodeSource(String nodeSourceName, String infrastructureType, Object[] infraParams,
+            String policyType, Object[] policyParams, boolean nodesRecoverable) {
         if (nodeSourceName == null) {
             throw new IllegalArgumentException("Node Source name cannot be null");
         }
+
+        logger.info("Define a node source : " + nodeSourceName);
+
         nodeSourceName = nodeSourceName.trim();
 
-        NodeSourceData nodeSourceData = new NodeSourceData(nodeSourceName,
-                                                           infrastructureType,
-                                                           infrastructureParameters,
-                                                           policyType,
-                                                           policyParameters,
-                                                           caller,
-                                                           nodesRecoverable);
-        boolean added = dbManager.addNodeSource(nodeSourceData);
+        NodeSourceData persistedNodeSource = new NodeSourceData(nodeSourceName,
+                                                                infrastructureType,
+                                                                infraParams,
+                                                                policyType,
+                                                                policyParams,
+                                                                caller,
+                                                                nodesRecoverable,
+                                                                NodeSourceStatus.NODES_UNDEPLOYED);
+
+        boolean added = dbManager.addNodeSource(persistedNodeSource);
 
         try {
-            return createNodeSource(nodeSourceData, nodesRecoverable);
+            validateNodeSourceNameOrFail(nodeSourceName);
+
+            NodeSource nodeSource = createNotActiveNodeSource(persistedNodeSource);
+
+            nodeSources.put(nodeSourceName, nodeSource);
+
+            this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_DEFINED,
+                                                                  caller.getName(),
+                                                                  nodeSourceName,
+                                                                  nodeSource.getDescription(),
+                                                                  caller.getName(),
+                                                                  nodeSource.getStatus().toString()));
+
+            logger.info("Node source " + nodeSourceName + " has been successfully defined by " + caller.getName());
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Node source " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)